consensus/core/src/dag_state.rs

```diff
@@ -803,22 +870,42 @@ impl DagState {
 
         // Clean up old cached data. After flushing, all cached blocks are guaranteed to be persisted.
         let mut total_recent_refs = 0;
-        for (authority_refs, last_committed_round) in self
-            .recent_refs
-            .iter_mut()
-            .zip(self.last_committed_rounds.iter())
-        {
-            while let Some(block_ref) = authority_refs.first() {
-                if block_ref.round
-                    <= Self::eviction_round(*last_committed_round, self.cached_rounds)
-                {
-                    self.recent_blocks.remove(block_ref);
-                    authority_refs.pop_first();
-                } else {
-                    break;
+
+        if self.gc_enabled() {
```

Comment:
It seems these two branches can also be merged by always setting evicted_rounds.